packages/react-components/source/react/library/drawer/Drawer.js

```diff
@@ -0,0 +1,98 @@
+import React, { useState } from 'react';
+import classnames from 'classnames';
+import {
+  bool,
+  string,
+  element,
+  node,
+  oneOf,
+  elementType,
+  func,
+} from 'prop-types';
+import Button from '../button';
+
+const propTypes = {
+  /** Selector with determines whether the drawer is open on first render, this prop will be overwritten by the open prop */
+  defaultOpen: bool,
+  /** Control whether the body content is open or closed with the open boolean prop. */
+  open: bool,
+  /** Callback which returns the new open prop value when the expand button is clicked */
+  buttonToggle: func,
```

Comment:
Can we use the common `on` prefix for callback handlers here? Something like `onToggle` would do.